hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -79,39 +81,52 @@ class DefaultSource extends RelationProvider
     val allPaths = path.map(p => Seq(p)).getOrElse(Seq()) ++ readPaths
 
     val fs = FSUtils.getFs(allPaths.head, sqlContext.sparkContext.hadoopConfiguration)
-    val globPaths = HoodieSparkUtils.checkAndGlobPathIfNecessary(allPaths, fs)
-
-    val tablePath = DataSourceUtils.getTablePath(fs, globPaths.toArray)
+    // Use the HoodieFileIndex only if the 'path' has specified with no "*" contains.
+    // And READ_PATHS_OPT_KEY has not specified.
+    // Or else we use the original way to read hoodie table.
+    val useHoodieFileIndex = path.isDefined && !path.get.contains("*") &&
```

Comment:
can we guard this feature itself with a data source option as well. i.e 

`val useHoodieFileIndex = hoodieFileIndexEnabled && path.isDefined && !path.get.contains("*") && ..`

Reply:
Hi @vinothchandar , Currently user query the hoodie table must specify some "*" at the path.  If the path has not specified the "*", the old way of query hoodie table may not work(exception or get no data). So we must use the  `HoodieFileIndex` for this case by default.